### PR TITLE
Support error threshold in the compute yaml

### PR DESCRIPTION
**Pull request by haozhestat**
In this PR, we add `parallel_error_threshold` into the dataclass of `pipeline_compute_config` so that the user could specify a unified error_threshold in the yaml file.

### CODE REVIEW

File: shrike/pipeline/pipeline_helper.py [hunk 1]
```diff
@@ -493,6 +498,9 @@ def _apply_parallel_runsettings(
             run_max_try=run_max_try
             if run_max_try is not None
             else self.config.compute.parallel_run_max_try,
+            error_threshold=error_threshold
```
Comment (dkmiller) [inline shrike/pipeline/pipeline_helper.py:501]: Can you add unit tests to verify that this works?

**Review by dkmiller — CHANGES_REQUESTED** [review]

File: shrike/pipeline/pipeline_config.py [hunk 2]
```diff
@@ -68,10 +68,11 @@ class pipeline_compute_config:  # pylint: disable=invalid-name
     hdi_conf: Optional[Any] = MISSING
 
     parallel_node_count: int = 10
-    parallel_process_count_per_node: int = 1
+    parallel_process_count_per_node: Optional[int] = MISSING
```
Comment (haozhestat) [inline shrike/pipeline/pipeline_config.py:71]: @evgri243 : I changed the default value of "parallel_process_count_per_node" from 1 to Null in this line.

File: shrike/pipeline/pipeline_helper.py (hunk 1 above)
Comment (haozhestat) [reply shrike/pipeline/pipeline_helper.py:505]: @dkmiller and @fuhuifang : I have added a unit test for `_apply_parallel_runsettings()` in the `tests\tests_pipeline\test_pipeline_helper.py` file. Could you take a look?

If this type of unit test looks good to you, I will create another PR to create unit test for all other functions in the `pipeline_helper.py`

**Review by dkmiller — APPROVED** [review]

File: tests/tests_pipeline/data/test_configuration.yaml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  parallel_error_threshold: -1
+  datatransfer_target: data-factory
+  compliant_datastore: workspaceblobstore
+  noncompliant_datastore: null
+
+module_loader:
+  use_local: '*'
+  force_default_module_version: null
+  force_all_module_version: null
+  local_steps_folder: ../../steps
```
Comment (fuhuifang) [inline tests/tests_pipeline/data/test_configuration.yaml:60]: Does this relative path work?

File: tests/tests_pipeline/test_pipeline_helper.py [hunk 4]
```diff
@@ -18,3 +24,64 @@ def test_validate_experiment_name():
...
+    """Unit tests for _apply_parallel_runsettings()"""
+    # Load the testing configuration YAML file
+    config = OmegaConf.load("tests/tests_pipeline/data/test_configuration.yaml")
+
+    # Initiate AMLPipelineHelper class with the testing configuration
+    pipeline_helper = AMLPipelineHelper(config=config)
+    pipeline_helper.connect()
+
+    # Create a module instance
+    module_instance_fun = Component._from_module_spec(
```
Comment (fuhuifang) [inline tests/tests_pipeline/test_pipeline_helper.py:42]: Can we use `pipeline_helper.component_load()` directly?

File: tests/tests_pipeline/data/test_configuration.yaml (hunk 3 above)
Comment (haozhestat) [reply tests/tests_pipeline/data/test_configuration.yaml]: Good catch. Addressed.

File: tests/tests_pipeline/test_pipeline_helper.py (hunk 4 above)
Comment (haozhestat) [reply tests/tests_pipeline/test_pipeline_helper.py]: Yes. I just updated it to `pipeline_helper.component_load()`, which is much simpler than and equivalent with `Component._from_module_spec()`

**Pull request merged**